mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/MemoryEventDeadLetters.java

```diff
@@ -74,14 +73,14 @@ public MemoryEventDeadLetters() {
         Preconditions.checkArgument(registeredGroup != null, REGISTERED_GROUP_CANNOT_BE_NULL);
 
         synchronized (deadLetters) {
-            return Flux.fromIterable(deadLetters.row(registeredGroup).keySet());
+            return Flux.fromIterable(ImmutableList.copyOf(deadLetters.row(registeredGroup).keySet()));
```

Comment:
You could synchronize only the creation of the immutable list.

Reply:
Not worth the pain of having a local variable/extracted function IMO

Reply:
I agree with @chibenwa on this